### ISSUE TITLE
Question/Clarrification - iptv-proxy behind VPN

### ISSUE THREAD

**Issue by twenty3thirty0** [pierre-emmanuelJ/iptv-proxy#154]
Hi.

I'm pretty sure the answer to this is "yes" but would just like to clarify....

If iptv-proxy is running on a machine with a VPN enabled... any IPTV client pointing to the iptv-proxy address will be served the content *through* the VPN... and i.e. not require the client to also connect to the VPN.

**Comment by NovalisVonHardenberg** [comment]
Thats possible with qmcgaw/gluetun
I can write a small tutorial tomorrow.

**Comment by ghost** [comment]
User docker-compose:
- One container, like Novalis wrote, with gluetun. I use it with nordvpn.
- In the compose from the proxy u can simply do: "network_mode: container:gluetun"

or u use the little advanced method with traefik to get a letsencrypt cert in case u need ssl.

**Comment by twenty3thirty0** [comment]
Thanks both.

From what i understand, gluetun would be useful for where I'd want the client to be on the same machine as the docker container (with iptv proxy and vpn running).

My scenario is a little simpler than that...
So i have a Windows box running iptv-proxy and VPN.
This works, and i can have any number of other machines and clients connected to it (on the same LAN). These work..even without a VPN enabled.
So my question is, are these VPN-less clients exposed as being connected indirectly to the source iptv provider. Or is iptv-proxy doing the job of getting the stream from the provider (via vpn) and serving it out locally, without any passed on connection to the source?

**Comment by ghost** [comment]
No. Gluetun is a vpn client. So u can start the gluetun container who etablish the connection to your vpn provider.
You can then let Docker route the traffic from another container trough it.
So in this case:    client (other machine) <--> iptv-proxy (win-box) <--> gluetun (win-box) <--> vpn-server <--> provider

Its just for security reason. With this setup u can start the proxy, but when gluetun is not running the proxy can't connect.
If you have your normal desktop vpn client not running and you start the proxy, then the provider will see the ip of the proxy.

The answer to your question is, YES.
Just the channel and movie pictures are not proxied, but the streams are.

For the provider it may still be possible to detect, that you are using different devices.
To prevent that you can set a static user-agent on in this line:
[xtreamHandles.go](https://github.com/pierre-emmanuelJ/iptv-proxy/blob/6561cba1e60cfd415fb6555b66524ba422a7aa09/pkg/server/xtreamHandles.go#L76)
Just replace   ctx.Request.UserAgent()  with a static string e.g.   "VLC/3554"   or   "IPTV Smarters/1.0.3 (iPhone; iOS 16.3; Scale/2.00)"

**Comment by twenty3thirty0** [comment]
Thanks @zimmflo thats been really really helpful.

As my vpn client is always on with a kill switch enabled already, i think im covered, and dont see much need for gluetun.

Makes sense that the epg channel list and media content is still direct.

Ill definitely look into the user-agent though. Thats a really good shout.

Thanks again

**Comment by mf1x** [comment]
@NovalisVonHardenberg 
Are you still keen to write a little guide?
It would be much appreciated.
